e2e/tests/components/button-spec.ts

```diff
@@ -0,0 +1,79 @@
+import { test, expect, devices } from '@playwright/test';
+
+test.describe('Button', () => {
+
+  let componentUrls: string[] = [];
+
+  const workingDirectory = process.cwd();
+  // split the last part of the workingDirectory path and add the storybook path
+  const localStorybookPath = `file://${workingDirectory.split('/').slice(0, -1).join('/')}/packages/react/storybook-static/index.html?path=/story/`;
+  const componentPath = `${localStorybookPath}components-button`;
+
+  [devices['Desktop Chrome'], devices['iPhone SE']].forEach(device => {
+    test(`button ${device.isMobile ? 'mobile' : 'desktop'}`, async ({ page }) => {
+      page.setViewportSize(device.viewport);
+      await page.goto(componentPath);
+
+      if (componentUrls.length == 0) {
+        // get components with data-parent-id="components-button"    
+        await expect(page.locator("#components-button")).toBeVisible();
+        const components = await page.locator('[data-parent-id="components-button"]').all();
+        // loop components and get each's url
+        for (const component of components) {
+          await component.getAttribute('href').then((href) => {
+            // don't add anything containing 'playground' to the list
+            if (href && !href.includes('playground')) {
+              componentUrls.push(href as string);
+            }
+          });
+        }
+      }
+    });
+
+    test(`subTest-${device.isMobile ? 'mobile' : 'desktop'}`, async ({ page }) => {
+      page.setViewportSize(device.viewport);
+      for (const componentUrl of componentUrls) {
+        await page.goto(`file://${componentUrl}`);
+        const iFrame = await page.frameLocator('iframe[title="storybook-preview-iframe"]');
+        const container = await iFrame.locator('body');
+
+        // name for the screenshot
+        const name = `${componentUrl.split('/').pop()}-${device.isMobile ? 'mobile' : 'desktop'}`;
+
+        // take screenshot, also waits for all to render
+        await expect(container).toHaveScreenshot(`${name}.png`);
+
+        const buttons = await container.locator('[data-playwright="true"]').all();
+
+        for (const [index, button] of buttons.entries()) {
+          await button.scrollIntoViewIfNeeded();
+          const buttonBoundingBox = await button.boundingBox() || { x: 0, y: 0, width: 0, height: 0 };
+          const outside = { x: buttonBoundingBox.x - 1, y: buttonBoundingBox.y - 1 };
+          const screenshotPrefix = `${name}-components-${index}`;
+
+          // to make sure nothing is focused
+          await page.mouse.move(outside.x, outside.y);
+          await page.mouse.down();
+          await page.mouse.up();
```

Comment:
When making the final code, it would be clearer to make helper functions like: 
clickOutside(element)
mouseDownInElementCenter(element)

And even:

takeScreenshotOfHoverState(element);

Reply:
Yeah, we could make a "unfocus" element helper for this.

Reply:
Added some helpers